packages/js/components/src/search/index.tsx

```diff
@@ -80,13 +218,15 @@ export class Search extends Component {
 				}
 				return this.props.autocompleter;
 			default:
-				return {};
+				throw new Error(
+					`No autocompleter found for type: ${ this.props.type }`
```

Comment:
I was thinking about that, too!

But actually, it's not a breaking change. I changed it to throw an error explicitly because TS reported a type error.

It implicitly throws an error given an invalid autocompleter type here.

https://github.com/woocommerce/woocommerce/blob/0ff14ce2b7005bd11b8f1dba937285e0f5d0fd4c/packages/js/components/src/search/index.tsx#L251-L252